preprocessing/nextclade/src/loculus_preprocessing/prepro.py

```diff
@@ -238,7 +238,74 @@ def run_sort(
     return alerts
 
 
-def enrich_with_nextclade(  # noqa: C901, PLR0912, PLR0914, PLR0915
+def assign_segment(
+    input_unaligned_sequences: dict[str, NucleotideSequence | None],
+    unaligned_nucleotide_sequences: dict[SegmentName, NucleotideSequence | None],
+    errors: list[ProcessingAnnotation],
+    aligned_nucleotide_sequences: dict[SegmentName, NucleotideSequence | None],
+    config: Config,
+):
+    # TODO: Add option to assign segment/subtype using nextclade sort
+    num_valid_segments = 0
+    num_duplicate_segments = 0
+    for segment in config.nucleotideSequences:
+        aligned_nucleotide_sequences[segment] = None
+        unaligned_segment = [
+            data
+            for data in input_unaligned_sequences
+            if re.match(segment + "$", data, re.IGNORECASE)
+        ]
+        if len(unaligned_segment) > 1:
+            num_duplicate_segments += len(unaligned_segment)
+            errors.append(
+                ProcessingAnnotation(
+                    unprocessedFields=[
+                        AnnotationSource(
+                            name=segment,
+                            type=AnnotationSourceType.NUCLEOTIDE_SEQUENCE,
+                        ),
+                    ],
+                    processedFields=[
+                        AnnotationSource(
+                            name=segment,
+                            type=AnnotationSourceType.NUCLEOTIDE_SEQUENCE,
+                        ),
+                    ],
+                    message="Found multiple sequences with the same segment name.",
+                )
+            )
+        elif len(unaligned_segment) == 1:
+            num_valid_segments += 1
+            unaligned_nucleotide_sequences[segment] = input_unaligned_sequences[
+                unaligned_segment[0]
+            ]
+        else:
+            unaligned_nucleotide_sequences[segment] = None
+    if len(input_unaligned_sequences) - num_valid_segments - num_duplicate_segments > 0:
+        errors.append(
+            ProcessingAnnotation(
+                unprocessedFields=[
+                    AnnotationSource(
+                        name="alignment",
+                        type=AnnotationSourceType.NUCLEOTIDE_SEQUENCE,
+                    ),
+                ],
+                processedFields=[
+                    AnnotationSource(
+                        name="alignment",
+                        type=AnnotationSourceType.NUCLEOTIDE_SEQUENCE,
+                    ),
+                ],
+                message=(
+                    "Found unknown segments in the input data - "
```

Comment:
lets improve this error message - and then prepro actually already validates the headers - we can already remove the validation in the backend if we fix how fasta and metadata are merged